src/main/java/duckling/responders/Responders.java

```diff
@@ -40,13 +38,11 @@ public Responders(
     }
 
     public Responder getResponder() {
-        Stream<Responder> candidates = responders.stream().filter(Responder::matches);
-        Optional<Responder> candidate = candidates.findFirst();
-
-        if (candidate.isPresent()) {
-            return candidate.get();
-        } else {
-            return new NotFound(request);
-        }
+        return responders.
+            stream().
+            filter(Responder::matches).
+            findFirst().
+            map((Responder candidate) -> candidate).
+            orElseGet(() -> new NotFound(request));
```

Comment:
Thanks!  

Oh, that reminds me - I tried to use `Function::identity` here and it was throwing an odd error. "Cannot convert Function<T,T> to U."  Does that mean it believes I'm trying to change types for some reason?